src/bun.js/webcore/S3File.zig

```diff
@@ -363,59 +363,58 @@ fn constructS3FileInternal(
 
 pub const S3BlobStatTask = struct {
     promise: JSC.JSPromise.Strong,
+    global: *JSC.JSGlobalObject,
     store: *Blob.Store,
+
     usingnamespace bun.New(S3BlobStatTask);
 
     pub fn onS3ExistsResolved(result: S3.S3StatResult, this: *S3BlobStatTask) void {
         defer this.deinit();
-        const globalThis = this.promise.globalObject().?;
         switch (result) {
             .not_found => {
-                this.promise.resolve(globalThis, .false);
+                this.promise.resolve(this.global, .false);
             },
             .success => |_| {
                 // calling .exists() should not prevent it to download a bigger file
                 // this would make it download a slice of the actual value, if the file changes before we download it
                 // if (this.blob.size == Blob.max_size) {
                 //     this.blob.size = @truncate(stat.size);
                 // }
-                this.promise.resolve(globalThis, .true);
+                this.promise.resolve(this.global, .true);
             },
             .failure => |err| {
-                this.promise.reject(globalThis, err.toJS(globalThis, this.store.data.s3.path()));
+                this.promise.reject(this.global, err.toJS(this.global, this.store.data.s3.path()));
             },
         }
     }
 
     pub fn onS3SizeResolved(result: S3.S3StatResult, this: *S3BlobStatTask) void {
         defer this.deinit();
-        const globalThis = this.promise.globalObject().?;
 
         switch (result) {
             .success => |stat_result| {
-                this.promise.resolve(globalThis, JSValue.jsNumber(stat_result.size));
+                this.promise.resolve(this.global, JSValue.jsNumber(stat_result.size));
             },
-            inline .not_found, .failure => |err| {
-                this.promise.reject(globalThis, err.toJS(globalThis, this.store.data.s3.path()));
+            .not_found, .failure => |err| {
+                this.promise.reject(this.global, err.toJS(this.global, this.store.data.s3.path()));
             },
         }
     }
 
     pub fn onS3StatResolved(result: S3.S3StatResult, this: *S3BlobStatTask) void {
         defer this.deinit();
-        const globalThis = this.promise.globalObject().?;
         switch (result) {
             .success => |stat_result| {
-                this.promise.resolve(globalThis, S3Stat.init(
+                this.promise.resolve(this.global, S3Stat.init(
```

Comment:
IMO This was better as a separate pointer copied since it's accessed multiple times here

Reply:
i can put these back on the stack where `const globalThis` was